fe/fe-core/src/main/java/org/apache/doris/job/scheduler/JobScheduler.java

```diff
@@ -0,0 +1,173 @@
+     */
+    private static final long BATCH_SCHEDULER_INTERVAL_MILLI_SECONDS = BATCH_SCHEDULER_INTERVAL_SECONDS * 1000L;
+
+    public void start() {
+        timerTaskScheduler = new HashedWheelTimer(new CustomThreadFactory("timer-task-scheduler"), 1,
+                TimeUnit.SECONDS, HASHED_WHEEL_TIMER_TICKS_PER_WHEEL);
+        timerTaskScheduler.start();
+        taskDisruptorGroupManager = new TaskDisruptorGroupManager();
+        taskDisruptorGroupManager.init();
+        this.timerJobDisruptor = taskDisruptorGroupManager.getDispatchDisruptor();
+        latestBatchSchedulerTimerTaskTimeMs = System.currentTimeMillis();
+        batchSchedulerTimerJob();
+        cycleSystemSchedulerTasks();
+    }
+
+    /**
+     * We will cycle system scheduler tasks every 10 minutes.
+     * Jobs will be re-registered after the task is completed
+     */
+    private void cycleSystemSchedulerTasks() {
+        log.info("re-register system scheduler timer tasks" + TimeUtils.longToTimeString(System.currentTimeMillis()));
+        timerTaskScheduler.newTimeout(timeout -> {
+            batchSchedulerTimerJob();
+            cycleSystemSchedulerTasks();
+        }, BATCH_SCHEDULER_INTERVAL_SECONDS, TimeUnit.SECONDS);
+
+    }
+
+    private void batchSchedulerTimerJob() {
+        executeTimerJobIdsWithinLastTenMinutesWindow();
+    }
+
+    public void scheduleOneJob(T job) {
+        if (!job.getJobStatus().equals(JobStatus.RUNNING)) {
+            return;
+        }
+        if (!job.getJobConfig().checkIsTimerJob()) {
+            //manual job will not scheduler
+            if (JobExecuteType.MANUAL.equals(job.getJobConfig().getExecuteType())) {
+                return;
+            }
+            //todo skip streaming job,improve in the future
+            if (JobExecuteType.INSTANT.equals(job.getJobConfig().getExecuteType()) && job.isReadyForScheduling()) {
+                schedulerImmediateJob(job);
+            }
+        }
+        //if it's timer job and trigger last window already start, we will scheduler it immediately
+        cycleTimerJobScheduler(job);
+    }
+
+    @Override
+    public void close() throws IOException {
+
+    }
+
+
+    private void cycleTimerJobScheduler(T job) {
+        List<Long> delaySeconds = job.getJobConfig().getTriggerDelayTimes(System.currentTimeMillis(),
+                System.currentTimeMillis(), latestBatchSchedulerTimerTaskTimeMs);
+        if (CollectionUtils.isNotEmpty(delaySeconds)) {
+            delaySeconds.forEach(delaySecond -> {
+                TimerJobSchedulerTask<T> timerJobSchedulerTask = new TimerJobSchedulerTask<>(timerJobDisruptor, job);
+                timerTaskScheduler.newTimeout(timerJobSchedulerTask, delaySecond, TimeUnit.SECONDS);
+            });
+        }
+    }
+
+
+    private void schedulerImmediateJob(T job) {
+        List<? extends AbstractTask> tasks = job.createTasks(TaskType.MANUAL);
+        if (CollectionUtils.isEmpty(tasks)) {
+            return;
```

Comment:
Need to update the job's status?
Or the job will be in RUNNING state forever

Reply:
oops...fixed